gamemodes/base/entities/entities/lua_run.lua

```diff
@@ -6,21 +6,62 @@ ENT.DisableDuplicator = true
 
 AccessorFunc( ENT, "m_bDefaultCode", "DefaultCode" )
 
+local ALLOWED_LUA, ALLOWED_MAPS = include( "cfc_block_luarun/config.lua" )
+local CURRENT_MAP_ALLOWED = ALLOWED_MAPS[game.GetMap()]
+
 function ENT:Initialize()
+
+	-- If the entity has its first spawnflag set, run the code automatically
+	if self:HasSpawnFlags( SF_LUA_RUN_ON_SPAWN ) then
+		self:RunCode( self, self, self:GetDefaultCode() )
+	end
+
 end
 
 function ENT:KeyValue( key, value )
+
+	if key == "Code" then
+		self:SetDefaultCode( value )
+	end
+
 end
 
 function ENT:SetupGlobals( activator, caller )
+
+	ACTIVATOR = activator
+	CALLER = caller
+
+	if IsValid( activator ) and activator:IsPlayer() then
+		TRIGGER_PLAYER = activator
+	end
+
 end
 
 function ENT:KillGlobals()
+
+	ACTIVATOR = nil
+	CALLER = nil
+	TRIGGER_PLAYER = nil
+
 end
 
 function ENT:RunCode( activator, caller, code )
+
+	if not CURRENT_MAP_ALLOWED or not ALLOWED_LUA[util.MD5( code )] then
```

Comment:
If the code is rejected maybe print the code and the hash in console so it's easy to find and add it to the whitelist.

Reply:
👍
Love this idea, consider it added.